code/datums/components/aiming.dm

```diff
@@ -0,0 +1,165 @@
+// Aiming component, ported from NSV
+
+/datum/component/aiming
+	can_transfer = FALSE
+	var/mob/living/user = null
+	var/mob/living/target = null
+	COOLDOWN_DECLARE(aiming_cooldown) // 5 second cooldown so you can't spam aiming for faster bullets/spamming lines
+
+/datum/component/aiming/Initialize(source)
+	. = ..()
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/stop_aiming)
+
+/datum/component/aiming/proc/aim(mob/user, mob/target)
+	set waitfor = FALSE // So we don't hold up the pointing animation.
+	if(!COOLDOWN_FINISHED(src, aiming_cooldown) || src.target || user == target) // No double-aiming
+		return
+	COOLDOWN_START(src, aiming_cooldown, 5 SECONDS)
+	src.user = user
+	src.target = target
+	user.visible_message("<span class='warning'>[user] points [parent] at [target]!</span>")
+	to_chat(target, "<span class='userdanger'>[user] is pointing [parent] at you! If you equip or drop anything they will be notified! \n <b>You can use *surrender to give yourself up</b>.</span>")
+	to_chat(user, "<span class='notice'>You're now aiming at [target]. If they attempt to equip anything you'll be notified by a loud sound.</span>")
+	playsound(target, 'sound/weapons/autoguninsert.ogg', 100, TRUE)
```

Comment:
This could use some balloon message to target

Reply:
I put one on the target, although it feels a bit redundant since the target already gets a radial popup and crosshairs over them